packages/core/src/crypto/jose/jwk/Ed25519Jwk.ts

```diff
@@ -0,0 +1,87 @@
+import type { JwkJson } from './Jwk'
+import type { Buffer } from '../../../utils'
+
+import { TypedArrayEncoder } from '../../../utils'
+import { KeyType } from '../../KeyType'
+import { JwaCurve, JwaKeyType } from '../jwa'
+import { JwaSignatureAlgorithm } from '../jwa/alg'
+
+import { Jwk } from './Jwk'
+import { hasKty, hasCrv, hasX, hasValidUse } from './validate'
+
+export class Ed25519Jwk extends Jwk {
+  public readonly x: string
+
+  public constructor({ x }: { x: string }) {
+    super()
+
+    this.x = x
+  }
+
+  public get kty() {
```

Comment:
we have a getter for the `kty` (key type) and `keyType`. I can imagine this causing some confusion, when to use which one. Do you think we should fix this or is it clear enough with the difference in name.

Reply:
Yup found it quite confusing as well, but didn't really know what to do. Any suggestions?

Reply:
Not too sure. The only thing I could come up with is to rename `kty` to `keyRepresentation`, but I would like `kty` to stay the same as its a direct mapping to the `JWA` defined key type.

I think we can leave this for now and maybe just add a docstring above `kty` that says that this is the key type for as defined in the `JWA` spec or IANA